### ISSUE TITLE
Cannot trigger handler with variable in name with ansible 1.9.4

### ISSUE THREAD

**Issue by talset** [ansible/ansible#13182]
Hi,

I tested to call an handler with a variable into notify like "restart {{ app }}" .
I found an closed issue similar : https://github.com/ansible/ansible/issues/12427
In this issue, @pmahoney replied it's working on 1.9.3 and it is correct also in 1.7.2. But in 1.9.4 it doesnt' work.

Here my test code : 

``` bash
---                                                                                                                                                                                                    
- hosts: masters
  remote_user: root
  sudo: yes
  vars:
    to_notify: "restart-foo"
  handlers:
    - name: restart-foo
      shell: echo "restart-foo"

  tasks:
    - name: ls
      shell: ls
      notify: "{{ to_notify }}"
```

Logs : 

``` bash
ansible-playbook post.yml 

PLAY [masters] **************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [master2.blue.foo.com]
ok: [master1.blue.foo.com]

TASK: [ls] ******************************************************************** 
changed: [master2.blue.foo.com]
changed: [master1.blue.foo.com]
ERROR: change handler ({{ to_notify }}) is not defined
```

Error : 

``` bash
ERROR: change handler ({{ to_notify }}) is not defined
```

Expected :

``` bash
Changed: change handler (restart-foo)
```

**Comment by gaelL** [comment]
Hi,

I tried from older version from git tags.

```
pip install git+git://github.com/ansible/ansible/@v1.7.2 
```

It works in `1.7.2` tags.
After that (`1.8.0` and other like `v1.9.3-1`) seems broken.

**Comment by bcoca** [comment]
there was a bug from 1.8 onwards that broke templating of name: in tasks, this now works again in 2.0

**Comment by bcoca** [comment]
closing this as it is fixed in 2.0 and it won't be backported
